stable_baselines/a2c/utils.py

```diff
@@ -158,6 +158,20 @@ def linear(input_tensor, scope, n_hidden, *, init_scale=1.0, init_bias=0.0):
         return tf.matmul(input_tensor, weight) + bias
 
 
+def apply_action_mask(input_tensor, mask_tensor, scope):
+    """
+    Perform element wise multiplication to mask invalid actions
+
+    :param input_tensor: (TensorFlow Tensor) The output layer of a policy
+    :param mask_tensor: (TensorFlow Tensor) A boolean tensor denoting which actions are valid
+    :param scope: (str) The TensorFlow variable scope
+    :return: (TensorFlow Tensor) The resulting tensor from element-wise multiplication between the input_tensor and
+    mask_tensor
+    """
+    with tf.variable_scope(scope):
+        return tf.add(input_tensor, mask_tensor, name="masking")
```

Comment:
Is that really an element wise multiplication?

Reply:
See: https://github.com/hill-a/stable-baselines/pull/453#issuecomment-537042170